tests/sync/partial_sync.cpp

```diff
@@ -0,0 +1,240 @@
+
+using namespace realm;
+
+struct TypeA {
+    size_t first_number;
+    size_t second_number;
+    std::string string;
+};
+
+struct TypeB {
+    size_t number;
+    std::string first_string;
+    std::string second_string;
+};
+
+enum class PartialSyncTestObjects { A, B };
+
+// Test helpers.
+namespace {
+
+Schema partial_sync_schema()
+{
+    return Schema{
+        {"partial_sync_object_a", {
+            {"first_number", PropertyType::Int},
+            {"second_number", PropertyType::Int},
+            {"string", PropertyType::String}
+        }},
+        {"partial_sync_object_b", {
+            {"number", PropertyType::Int},
+            {"first_string", PropertyType::String},
+            {"second_string", PropertyType::String},
+        }}
+    };
+}
+
+void populate_realm(Realm::Config& config, std::vector<TypeA> a={}, std::vector<TypeB> b={})
+{    
+    auto r = Realm::get_shared_realm(config);
+    r->begin_transaction();
+    {   
+        const auto& object_schema = *r->schema().find("partial_sync_object_a");
+        const auto& first_number_prop = *object_schema.property_for_name("first_number");
+        const auto& second_number_prop = *object_schema.property_for_name("second_number");
+        const auto& string_prop = *object_schema.property_for_name("string");
+        TableRef table = ObjectStore::table_for_object_type(r->read_group(), "partial_sync_object_a");
+        for (auto& current : a) {
+            size_t row_idx = sync::create_object(r->read_group(), *table);
+            table->set_int(first_number_prop.table_column, row_idx, current.first_number);
+            table->set_int(second_number_prop.table_column, row_idx, current.second_number);
+            table->set_string(string_prop.table_column, row_idx, current.string);
+        }
+    }
+    {
+        const auto& object_schema = *r->schema().find("partial_sync_object_b");
+        const auto& number_prop = *object_schema.property_for_name("number");
+        const auto& first_string_prop = *object_schema.property_for_name("first_string");
+        const auto& second_string_prop = *object_schema.property_for_name("second_string");
+        TableRef table = ObjectStore::table_for_object_type(r->read_group(), "partial_sync_object_b");
+        for (auto& current : b) {
+            size_t row_idx = sync::create_object(r->read_group(), *table);
+            table->set_int(number_prop.table_column, row_idx, current.number);
+            table->set_string(first_string_prop.table_column, row_idx, current.first_string);
+            table->set_string(second_string_prop.table_column, row_idx, current.second_string);
+        }
+    }
+    r->commit_transaction();
+    // Wait for uploads
+    std::atomic<bool> upload_done(false);
+    auto session = SyncManager::shared().get_existing_active_session(config.path);
+    session->wait_for_download_completion([&](auto) { upload_done = true; });
+    EventLoop::main().run_until([&] { return upload_done.load(); });
```

Comment:
Hmm, isn't this effectively a busy loop?

Reply:
This is unit test code intended to stop execution of the test until all the changes are ready. It won’t affect the bindings in any way.

Reply:
I understand, it just seems wasteful. If there is no easy way to avoid it, no big problem.

Reply:
We currently have only a very minimal runloop abstraction available in the object store tests, so this is the best we can do.